staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -95,6 +95,39 @@ func TestUpdateOnlyDoesNotTrackManagedFields(t *testing.T) {
 	}
 }
 
+func TestUpdateApplyConflict(t *testing.T) {
+	f := NewTestFieldManager()
+
+	obj := &corev1.Pod{}
+	obj.ObjectMeta.ManagedFields = []metav1.ManagedFieldsEntry{{}}
+
+	obj2 := obj.DeepCopy()
+	obj2.ObjectMeta.Labels = map[string]string{"k": "v"}
+
+	newObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	newObj.SetAPIVersion("apps/v1")
+	newObj.SetKind("Deployment")
+	newObj.SetName("deployment")
+
+	oldObject, err := f.Update(obj, newObj, "fieldmanager_test")
```

Comment:
I also tried passing obj2 as 2nd argument.
The assertion fails in both cases.

Passing obj as 1st argument to f.Apply doesn't change the result.